pkg/proxy/creator/creator.go

```diff
@@ -0,0 +1,236 @@
+package creator
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/process"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-chain-proxy-go/api"
+	"github.com/multiversx/mx-chain-proxy-go/common"
+	"github.com/multiversx/mx-chain-proxy-go/config"
+	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/multiversx/mx-chain-proxy-go/metrics"
+	"github.com/multiversx/mx-chain-proxy-go/observer"
+	processProxy "github.com/multiversx/mx-chain-proxy-go/process"
+	"github.com/multiversx/mx-chain-proxy-go/process/cache"
+	"github.com/multiversx/mx-chain-proxy-go/process/database"
+	processFactory "github.com/multiversx/mx-chain-proxy-go/process/factory"
+	versionsFactory "github.com/multiversx/mx-chain-proxy-go/versions/factory"
+	proxy2 "github.com/multiversx/mx-chain-simulator-go/pkg/proxy"
+)
+
+var log = logger.GetOrCreate("proxy")
+
+// ArgsProxy holds the arguments needed to create a new instance of proxy
+type ArgsProxy struct {
+	PathToConfig string
+	Config       *config.Config
+	NodeHandler  process.NodeHandler
+}
+
+type proxy struct {
+	closableComponents *data.ClosableComponentsHandler
+	httpServer         *http.Server
+}
+
+// CreateProxy will create a new instance of proxy
+func CreateProxy(args ArgsProxy) (proxy2.ProxyHandler, error) {
```

Comment:
with a little bit of code refactoring in the proxy repo, we can reuse this construction part. Not for this PR.

Reply:
yes, will make a task to refactor the proxy main.go file